### PR TITLE
Repair tools

### PR DESCRIPTION
**Pull request by benwr**
@hwchen

a couple of trivial standalone tools you can use to check and repair the database

### CODE REVIEW

File: src/bin/repair_taxonomy_db.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+fn main() -> Result<()> {
+    let args = Args::parse();
+
+    let mut config = TaxonomyDatabaseConfig::new().cache_size(10_000_000_000);
+    if let Some(tax_dir) = args.taxonomy_dir {
+        config = config.location(tax_dir);
+    }
+
+    let db = config.build()?;
```
Comment (hwchen) [inline src/bin/repair_taxonomy_db.rs:25]: Won't this rebuild the whole db before repairing? I think that this binary is supposed to just repair.

**Review by hwchen — APPROVED** [review]
Ah, I see that whether or not it builds a new db depends on configuration options. I think this is a bit confusing, I'd expect both an `open` and a `build` action. But that doesn't block this PR.

File: src/bin/repair_taxonomy_db.rs (hunk 1 above)
Comment (hwchen) [reply src/bin/repair_taxonomy_db.rs:25]: ah, nvm

**Comment by benwr** [comment]
Yeah I agree that this isn't optimal; I originally wrote this tool as a quick hack :/

**Comment by benwr** [comment]
Oh, I see what you mean. Yeah in this case `build` is meant in the sense of `builder pattern`, not `build the database`. Is there a better word for that?

**Comment by hwchen** [comment]
I would probably have `build_new` and `open` as different finalizers for the builder pattern. And then just skip using `build` as a finalizer, since it's an overloaded term here. Or maybe not have a builder pattern, I'm not sure that there's enough options that it's necessary. 

For this amount of code, I don't worry much about that structure, but perhaps would at least be nice to have an explanatory comment on `build` that it does both `build_new` and `open`.

**Pull request merged**